core/include/gnuradio-4.0/Settings.hpp

```diff
@@ -526,8 +560,11 @@ class CtxSettings : public SettingsBase {
 #endif
         }
 
-        const std::optional<SettingsCtx> bestMatchSettingsCtx = findBestMatchSettingsCtx(ctx);
-        if (!bestMatchSettingsCtx || bestMatchSettingsCtx == _activeCtx) {
+        std::optional<SettingsCtx> bestMatchSettingsCtx = findBestMatchSettingsCtx(ctx);
+        if (!bestMatchSettingsCtx) {
+            _storedParameters[ctx.context] = {};
+            bestMatchSettingsCtx           = ctx;
+        } else if (bestMatchSettingsCtx == _activeCtx) {
```

Comment:
I believe this code can be removed. Since the method returns a `std::optional`, it will return `std::nullopt` if `ctx` is not found. Additionally, the default context is always present and is initialized in the `init()` function. Therefore, extra checks are unnecessary.